runtime/drivers/duckdb/duckdb.go

```diff
@@ -48,6 +50,15 @@ func (d Driver) Open(dsn string, logger *zap.Logger) (drivers.Connection, error)
 		return nil
 	})
 	if err != nil {
+		// Adding check if using old database files with upgraded version
+		if strings.Contains(err.Error(), "Trying to read a database file with version number") {
+			return nil, fmt.Errorf("The database file was created with an old version of Rill. Please remove %q and restart", cfg.DSN)
```

Comment:
Refactor message to: `"database file %q was created with an older, incompatible version of Rill (please remove it and try again)"`